spec/Section 6 -- Execution.md

```diff
@@ -734,9 +734,22 @@ the {objectValue}.
 
 ResolveAbstractType(abstractType, objectValue):
 
-- Return the result of calling the internal method provided by the type system
-  for determining the Object type of {abstractType} given the value
-  {objectValue}.
+- Let {resolvedType} be the result of calling the internal method provided by
+  the type system for determining the Object type of {abstractType} given the
+  value {objectValue}.
+- Assert {resolvedType} is an Object type.
+- If {IsSubType(abstractType, resolvedType)}:
+  - Return {resolvedType}.
+- Otherwise, raise a field error.
```

Comment:
This should also be an assert, I think. "determining the Object type of {abstractType} given the value {objectValue}" already specifies it's an object type, and that it's a valid object type within {abstractType}; so the asserts should just underline this.

```suggestion
- Assert {IsSubType(abstractType, resolvedType)}.
- Return {resolvedType}.
```

Reply:
I'm not sure "sub type" is the right terminology here; an object type isn't really a "sub type" of a union, it's a member. I'm not sure what the right terminology is though, I've had a quick scan through the spec and couldn't find a suitable term. I guess it's okay to say an object type that implements an interface is effectively a member of the interface, so let's use member for now until a better term appears:

```suggestion
- Assert {IsAbstractMemberType(abstractType, resolvedType)}.
- Return {resolvedType}.
```